packages/generator/ni_measurement_plugin_sdk_generator/client/_support.py

```diff
@@ -208,7 +208,7 @@ def get_configuration_parameters_with_type_and_default_values(
             if metadata.repeated:
                 formatted_value = ", ".join(f"Path({repr(value)})" for value in default_value)
                 default_value = f"[{formatted_value}]"
-                parameter_type = f"List[{parameter_type}]"
+                parameter_type = f"Iterable[{parameter_type}]"
```

Comment:
Similar to Iterable for input vs. Sequence for output, we really ought to accept PurePath for input and return Path for output.

Reply:
I changed the `measure()` and `stream_measure()` methods to accept `pathlib.PurePath` and return `pathlib.Path`. These methods don't actually access the filesystem, they just send the paths across the network.

I didn't change `register_pin_map` because it does actually access the filesystem. (It reads the pin map contents and then sends the contents across the network.)